src/sftp.rs

```diff
@@ -487,7 +491,7 @@ impl<'sftp> Write for File<'sftp> {
             raw::libssh2_sftp_write(self.raw, buf.as_ptr() as *const _, buf.len() as size_t)
         };
         if rc < 0 {
-            Err(io::Error::new(ErrorKind::Other, self.sftp.last_error()))
+            Err(Error::from_errno(rc as i32).into())
```

Comment:
sorry for not noticing this earlier: this `i32` should probably also be `_` as above